docs/UserGuide.md

```diff
@@ -126,118 +176,294 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
+
+Marks a student as absent for a lesson.
+
+Format: `unmarkstudent li/LESSON_INDEX si/STUDENT_INDEX`
+
+* `LESSON_INDEX` refers to the index number shown in the displayed list of tasks.
+* `STUDENT_INDEX` refers to the index number shown in the student list for the specified lesson. (use `listlesson` to view the student list for lessons)
+* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​
+
+Example:
+* `unmarkstudent li/2 si/3` marks the 3rd student in the 2nd lesson as absent.
+
+### Adding a note to a lesson : `addnote`
+
+Adds a note for a lesson or for a specific student in a lesson.
+
+Format: `addnote n/NOTE li/LESSON INDEX [si/STUDENT_INDEX]`
+
+* si/STUDENT_INDEX is optional, if specified the command will add to corresponding student notes.
+
+* `LESSON_INDEX` refers to the index number shown in the displayed list of tasks.
+* `STUDENT_INDEX` refers to the index number shown in the student list for the specified lesson. (use `listlesson` to view the student list for lessons)
+* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​
+
+Example:
+* `addnote n/Get back to jeff on streams li/1 si/2` adds a note for the 2nd student in the 1st lesson.
+* `addnote n/Get back to the class on streams li/1` adds a note for the 1st lesson.
+
+### Deleting a note from a lesson : `deletenote`
+
+Deletes a note for a lesson or for a specific student in a lesson.
+
+Format: `deletenote ni/NOTE INDEX li/LESSON INDEX [si/STUDENT_INDEX]`
+
+* si/STUDENT_INDEX is optional, if specified the command will delete from the corresponding student notes.
+
+* `NOTE_INDEX` refers to the index number shown in the displayed list of notes for a lesson or for a specific student in a lesson. (use `listlesson` to view the note lists for lessons)
+* `LESSON_INDEX` refers to the index number shown in the displayed list of tasks.
+* `STUDENT_INDEX` refers to the index number shown in the student list for the specified lesson. (use `listlesson` to view the student list for lessons)
+* `NOTE_INDEX`, `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​
+
+Example:
+* `deletenote ni/1 li/1 si/2` deletes a note for the 2nd student from the 1st lesson.
+* `deletenote ni/1 li/1` deletes a note from the 1st lesson.
+
+### Adding participation for a student in a lesson : `addpart`
+
+Adds participation for a student in a lesson.
+
+Format: `addpart p/PARTICIPATION li/LESSON INDEX si/STUDENT_INDEX`
+
+* `PARTICIPATION` **must be a positive integer** from 0 to 500.
+* `LESSON_INDEX` refers to the index number shown in the displayed list of tasks.
+* `STUDENT_INDEX` refers to the index number shown in the student list for the specified lesson. (use `listlesson` to view the student list for lessons)
+* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​
+
+Example:
+* `addpart p/100 li/1 si/2` sets participation for the 2nd student from the 1st lesson to be 100.
```

Comment:
i think we should specify that the `Lesson` here must be a `Studio`, if for example `Mastery Check` is provided as LESSON_INDEX, the command will not work. 
```suggestion
### Adding participation for a student in a lesson : `addpart`

Adds participation for a student in a lesson.

Format: `addpart p/PARTICIPATION li/LESSON_INDEX si/STUDENT_INDEX`

* `PARTICIPATION` **must be a positive integer** from 0 to 500.
* `LESSON_INDEX` refers to the index number shown in the displayed list of tasks.
* `STUDENT_INDEX` refers to the index number shown in the student list for the specified lesson. (use `listlesson` to view the student list for lessons)
* `LESSON_INDEX` and `STUDENT_INDEX` **must be a positive integer** 1, 2, 3, …​

Example:
* `addpart p/100 li/1 si/2` sets participation for the 2nd student from the 1st lesson to be 100.
```